src/insexpand.c

```diff
@@ -3538,7 +3538,8 @@ get_next_filename_completion(void)
     char_u	**sorted_matches;
     int		*fuzzy_indices_data;
 
-    if (in_fuzzy)
+    if (in_fuzzy && (vim_strchr(leader, '/') == NULL
```

Comment:
1) Add a comment
2) I'd expect the fuzzy matching to be applied to the trailing part of the path.

Reply:
I am also thinking about point 2. So here we should get the final pattern part more carefully then do fuzzy